plugins/modules/katello_manifest.py

```diff
@@ -119,13 +119,15 @@ def main():
                     params['repository_url'] = entity_dict['repository_url']
                 params.update(scope)
                 changed, result = module.resource_action('subscriptions', 'upload', params, files=files)
-                for error in result['humanized']['errors']:
-                    if "same as existing data" in error:
-                        changed = False
-                    elif "older than existing data" in error:
-                        module.fail_json(msg="Manifest is older than existing data.")
-                    else:
-                        module.fail_json(msg="Upload of the manifest failed: %s" % error)
+                # This is not completely accurate, because we cannot be sure something changed until we attempt the actual upload
```

Comment:
Pain point number one.

Reply:
I think this block lies about the state of the manifest in Katello and we just should say that this module does not support check mode?

we have https://github.com/theforeman/foreman-ansible-modules/issues/172 open and I think we can't fix it until Katello fixes their API.

cc @akofink @sean797 for their knowledge

Reply:
Fine with just _not_ stating to support checkmode.

Reply:
Yep, Katello needs to fix their API first. +1 to stating no support of checkmode or not stating support, either one.

Reply:
No check_mode it is...
Funny thing: `AnsibleModule`, when called with `check_mode` despite it not beeing stated as supported calls `self.exit_json` from its initializer to state `changed: false`.
This broke our derived modules, because we overloaded `exit_json` in a way that it expects the initializer to have run.
Does anyone think, this is an ansible bug?